### ISSUE TITLE
EvtLocalAddressesUpdated event is difficult to use correctly

### ISSUE THREAD

**Issue by Stebalien** [libp2p/go-libp2p#863]
For implementation convenience (I think?), `EvtLocalAddressesUpdated` doesn't _require_ that the announced addresses include diff information. Unfortunately, that means a robust application needs to implement address diffing logic even though go-libp2p always sets the `Diffs` field to true.

**Comment by aarshkshah1992** [comment]
@yusefnapora Is there any reason setting the `Diff` field is optional. Given that we have ONLY one emitter for that event (`Host`) who does produce a `Diff`, can we make this mandatory and incur the loss of generality ?

I think Sir Steb has a point. It is a bit awkward to use this event knowing the Diff might be false.

**Comment by Stebalien** [comment]
I'm also fine just removing it entirely and telling the user "something has changed, figure it out".

**Comment by aarshkshah1992** [comment]
ping @yusefnapora @raulk. 

Hey guys,

Do you have any concerns with this ?

**Comment by yusefnapora** [comment]
I'm fine with either making diffs mandatory or even removing them entirely, since we're not actually consuming the diffs anywhere yet.

If we want to later extend identify/delta to include addresses as well as protocol changes we should probably keep the diffs and make them mandatory. Otherwise removing them seems simpler.

**Comment by Stebalien** [comment]
> If we want to later extend identify/delta to include addresses as well as protocol changes we should probably keep the diffs and make them mandatory. Otherwise removing them seems simpler.

I don't think we _can_ include address changes once we switch to signed address records.
